jsk_perception/scripts/train_ssd.py

```diff
@@ -0,0 +1,214 @@
+        self.mean = mean
+
+    def __call__(self, in_data):
+        # There are five data augmentation steps
+        # 1. Color augmentation
+        # 2. Random expansion
+        # 3. Random cropping
+        # 4. Resizing with random interpolation
+        # 5. Random horizontal flipping
+
+        img, bbox, label = in_data
+
+        # 1. Color augmentation
+        img = random_distort(img)
+
+        # 2. Random expansion
+        if np.random.randint(2):
+            img, param = transforms.random_expand(
+                img, fill=self.mean, return_param=True)
+            bbox = transforms.translate_bbox(
+                bbox, y_offset=param['y_offset'], x_offset=param['x_offset'])
+
+        # 3. Random cropping
+        img, param = random_crop_with_bbox_constraints(
+            img, bbox, return_param=True)
+        bbox, param = transforms.crop_bbox(
+            bbox, y_slice=param['y_slice'], x_slice=param['x_slice'],
+            allow_outside_center=False, return_param=True)
+        label = label[param['index']]
+
+        # 4. Resizing with random interpolatation
+        _, H, W = img.shape
+        img = resize_with_random_interpolation(img, (self.size, self.size))
+        bbox = transforms.resize_bbox(bbox, (H, W), (self.size, self.size))
+
+        # 5. Random horizontal flipping
+        img, params = transforms.random_flip(
+            img, x_random=True, return_param=True)
+        bbox = transforms.flip_bbox(
+            bbox, (self.size, self.size), x_flip=params['x_flip'])
+
+        # Preparation for SSD network
+        img -= self.mean
+        mb_loc, mb_label = self.coder.encode(bbox, label)
+
+        return img, mb_loc, mb_label
+
+
+def main():
+    rospack = rospkg.RosPack()
+    jsk_perception_datasets_path = osp.join(
+        rospack.get_path('jsk_perception'), 'learning_datasets')
+
+    parser = argparse.ArgumentParser()
+    # Dataset directory
+    parser.add_argument('--train-dataset-dir', type=str,
+                        default=osp.join(jsk_perception_datasets_path,
+                                         'kitchen_dataset', 'train'))
+    parser.add_argument('--val-dataset-dir', type=str,
+                        default=osp.join(jsk_perception_datasets_path,
+                                         'kitchen_dataset', 'test'))
+    parser.add_argument(
+        '--model-name', choices=('ssd300', 'ssd512'), default='ssd512')
+    parser.add_argument('--gpu', type=int, default=0)
+    parser.add_argument('--batch-size', type=int, default=8)
+    parser.add_argument('--max-epoch', type=int, default=100)
+    parser.add_argument('--out-dir', type=str, default=None)
+    args = parser.parse_args()
+
+    train_dataset = DetectionDataset(args.train_dataset_dir)
+    fg_label_names = train_dataset.fg_class_names
+
+    if args.model == 'ssd300':
+        model = SSD300(
+            n_fg_class=len(fg_label_names),
+            pretrained_model='imagenet')
+    elif args.model == 'ssd512':
```

Comment:
`args.model_name`, or maybe set a `dest` in the argument?

Reply:
Thx, I updated

Reply:
thx, i updated